packages/harmony/src/components/input/TextInput/TextInput.mdx

```diff
@@ -0,0 +1,240 @@
+  Canvas,
+  Meta,
+  Unstyled,
+  Title,
+  Subtitle,
+  Description,
+  Controls,
+  Primary
+} from '@storybook/blocks'
+
+import { Flex } from 'components'
+import { IconSearch } from 'icons'
+import { ComponentRules, RelatedComponents } from 'storybook/components'
+
+import { TextInput } from './TextInput'
+import * as TextInputStories from './TextInput.stories'
+import { TextInputSize } from './types'
+
+<Meta of={TextInputStories} />
+
+<Title />
+
+- [Overview](#overview)
+- [Props](#props)
+- [Configuration options](#configuration-options)
+- [Use cases and examples](#use-cases-and-examples)
+- [Do's and dont's](#dos-and-donts)
+- [Related Components](#related-components)
+
+## Overview
+
+<Subtitle />
+<Description />
+<Primary />
+
+## Props
+
+<Controls />
+
+## Configuration Options
+
+### Sizes
+
+<Canvas of={TextInputStories.Sizes} />
+
+### States
+
+<Canvas of={TextInputStories.DefaultStates} />
+<Canvas of={TextInputStories.SmallStates} />
+
+### Leading and trailing icons
+
+<Canvas of={TextInputStories.Icons} />
+
+### Assistive Text
+
+<Canvas of={TextInputStories.AssistiveText} />
+
+### Validation and counter
+
+Use validation to give feedback when an invalid input is provided. The validation error should appear when the user is done typing and out of focus.
+
+<Canvas of={TextInputStories.Validation} />
+
+### Special Inputs
+
+An input in which the trailing and leading icons are replaced with type (Title-large) and are positioned below the input’s label.
+
+<Canvas of={TextInputStories.SpecialInputs} />
+
+## Use cases and examples
+
+- Always use placeholder in input fields
+- Character or word counters should be used if there is a limit on the input.
+- Use icons to help users identify what the input is asking for or to turn confiscation on and off
+- Always use a string or icon to label the input.
+
+## Do's and don'ts
+
+<ComponentRules
+  rules={[
+    {
+      positive: {
+        description: 'Place inputs 16px apart at minimum.',
+        component: (
+          <Unstyled style={{ width: '100%' }}>
+            <Flex w='100%' direction='column' gap='l'>
+              <TextInput label='Display Name' />
+              <TextInput label='@Handle' />
+            </Flex>
+          </Unstyled>
+        )
+      },
+      negative: {
+        description: 'Don’t place inputs 8px apart.',
+        component: (
+          <Unstyled style={{ width: '100%' }}>
+            <Flex w='100%' direction='column' gap='s'>
+              <TextInput label='Display Name' />
+              <TextInput label='@Handle' />
+            </Flex>
+          </Unstyled>
+        )
+      }
+    },
+    {
+      positive: {
+        description: 'Use sentence case for labels and placeholder text.',
+        size: 'small',
+        component: (
+          <Unstyled style={{ width: '100%' }}>
+            <TextInput
+              label='Amount to send'
+              placeholder='Enter an amount'
+              endAdornmentText='$AUDIO'
```

Comment:
add $ start adornment for consistency with the Don't

Reply:
oh jk it's an $AUDIO input

Still maybe want to use the same input on both sides?

Reply:
Oh I didn't think that AUDIO is supposed to have the $ start adornment? I can double check with Sammie to see